src/patchelf.cc

```diff
@@ -1878,10 +1878,22 @@ static void patchElf2(ElfFile && elfFile, const FileContents & fileContents, con
 
     if (printNeeded) elfFile.printNeededLibs();
 
-    elfFile.removeNeeded(neededLibsToRemove);
-    elfFile.replaceNeeded(neededLibsToReplace);
-    elfFile.addNeeded(neededLibsToAdd);
-    elfFile.clearSymbolVersions(symbolsToClearVersion);
+    if (!neededLibsToRemove.empty()) {
+        elfFile.removeNeeded(neededLibsToRemove);
+        elfFile.rewriteSections();
```

Comment:
It might be better if those functions would call `rewriteSections` than and handle the empty check...

Reply:
done.